crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -202,6 +202,36 @@ def openstack_check
           )
           ret[:lbaas_v1] = true unless out[:stdout].nil? || out[:stdout].empty?
         end
+      end
+
+      def mixed_roles_check
+        conflicting_roles = [
+          "cinder-controller",
```

Comment:
Sounds like interesting idea, but I'm afraid some of these dependencies are upgrade only related...